src/Analyzers/Core/CodeFixes/MakeDeclarationPartial/AbstractMakePartialCodeFixProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.CodeAnalysis.MakeDeclarationPartial
+{
+    internal abstract class AbstractMakeDeclarationPartialCodeFixProvider<TDeclarationSyntax> : CodeFixProvider
+        where TDeclarationSyntax : SyntaxNode
+    {
+        protected abstract string GetDeclarationName(TDeclarationSyntax node);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var declarationNode = (TDeclarationSyntax)syntaxRoot.FindNode(context.Span);
```

Comment:
this cast is not safe.  we don't know where the span of htis diagnostic may be (compiler is free to move it aroun).  this needs to do an 'as' check and bail if it's null.